### PR TITLE
AP-2176 Asssess each proceeding type separately

### PR DESCRIPTION
**Pull request by stephenrichards**
## Assess each proceeding type separately

The original ticket called for assessing each proceeding type separately, and producing output in version 4 format.  I have split this ticket into two -this covers just the assessment part, and there will be another ticket to generate version 4 output if version 4 requested.

### Changes
- For version 3 requests, where no proceeding type is specified, populated the `proceeding_type_codes` field on the `Assessment` record with a dummy domestic abuse proceeding type code (`DA001`)
- Created a new table `eligibilities` to hold eligibility at a proceeding type level, keyed by assessment_id and proceeding_type_code.  The models held in this table are:
  -  `Eligibility::Capital`
  - `Eligibility::GrossIncome`
  - `Eligibility::DisposableIncome`
- Added an `eligibilities` association to the `CapitalSummary`, `GrossIncomeSummary` and `DisposableIncomeSummary` to link the appropriate `Eligibility` record to the summary.
- Wrote a creator class for each type of eligibility record to create a record for each proceeding type and populate it with the appropriate thresholds
- Added method `#summarized_assessment_result` to provide one summarized result taking into account the assessment results on each eligibility record.  This replaces the exisiting `assessment result` column on the summary records which will be removed in a later ticket.
  

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2176)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [ ] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [ ] The PR description should say what you changed and why, with a link to the JIRA story.
- [ ] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/services/assessors/capital_assessor.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
     end
 
     private
 
-    def result
-      return :eligible if assessed_capital <= lower_threshold
-
-      :contribution_required
+    def summary_result
+      # require Rails.root.join('app/services/utilities/result_summarizer')
```
Comment (willc-work) [inline app/services/assessors/capital_assessor.rb:13]: can this comment be removed now?

File: app/services/workflows/self_employed_workflow.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module Workflows
   class SelfEmployedWorkflow < BaseWorkflowService
     def call
-      raise 'Not yet implemented: Check Fincancial Eligibility service currently does not handle self-employed applicants'
+      raise 'Not yet implemented: Check Fincancial Base service currently does not handle self-employed applicants'
```
Comment (willc-work) [inline app/services/workflows/self_employed_workflow.rb:4]: Nit pick - Financial is spelled incorrectly.
Are we renaming the application for a specific reason?

Comment (stephenrichards) [reply app/services/workflows/self_employed_workflow.rb]: Nope - that was a bug.  Fixed now

File: spec/models/gross_income_summary_spec.rb [hunk 3]
```diff
@@ -37,25 +37,25 @@
...
-      gross_income_summary.reload
-    end
-
-    it 'persists the data' do
-      data.each do |method, value|
-        expect(gross_income_summary.__send__(method).to_d).to eq(value.to_d)
-      end
-    end
-  end
+  # describe '#summarise!' do
```
Comment (willc-work) [inline spec/models/gross_income_summary_spec.rb:40]: can these tests be removed?

**Review by willc-work — APPROVED** [review]
Left some comments about comments in code/commented out tests and. renaming but nothing major

File: spec/services/workflows/self_employed_workflow_spec.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@ module Workflows
       it 'raises' do
         expect {
           described_class.call(assessment)
-        }.to raise_error RuntimeError, 'Not yet implemented: Check Fincancial Eligibility service currently does not handle self-employed applicants'
+        }.to raise_error RuntimeError, 'Not yet implemented: Check Financial ELigibility service currently does not handle self-employed applicants'
```
Comment (willc-work) [inline spec/services/workflows/self_employed_workflow_spec.rb:10]: Eligibility has a capital L in it

**Pull request merged**